website/cue/reference/components/sources/datadog_agent.cue

```diff
@@ -92,6 +92,19 @@ components: sources: datadog_agent: {
 			required:    false
 			type: bool: default: true
 		}
+		trace_proto: {
+			common:      false
+			description: "Configure which trace format protocol the source shall accept."
+			required:    false
+			type: string: {
+				default: "v1v2"
+				enum: {
+					v1:   "Only support traces coming from older Datadog trace-agent (before v7.33)"
+					v2:   "Only support traces coming from newer Datadog trace-agent (v7.33 and after)"
+					v1v2: "Support both newer and older forma."
+				}
```

Comment:
What advantage do users got by having this be configurable over just accepting both?

Reply:
I thought about one single use case to filter out older agents, but indeed this could option can be completely dropped, or exchanged in favour of a single boolean option like `accept_old_format`.  I don't have a strong opinion on that.

Reply:
👍 I'm inclined to just drop the option and have the source always accept both until someone asks for it.

Reply:
I'm of the opinion that we should drop the option to accept both.

The endpoint that accepts both will parse as V2 first and if that errors then parse as V1. I'm not keen on the performance implications if a lot of events come in as V1.

It looks like the way DataDog itself works is by providing separate endpoints for V1 and V2. We could emulate this by forcing a source to be V1 or V2, and then if a user needed to accept both versions they could create separate sources (albeit accepting on different ports rather than paths) for each version. This encourages performant configuration.

We could default to V2.

Reply:
Aha, great! That seems like a good approach then, that would let us keep accepting both formats without the performance penalty.